tests/31sync/02sync.pl

```diff
@@ -1,3 +1,222 @@
+use Future::Utils qw( repeat );
+
+push our @EXPORT, qw(
+   matrix_do_and_wait_for_sync
+   matrix_send_room_text_message_and_wait_for_sync
+   matrix_send_room_message_and_wait_for_sync
+   matrix_create_room_and_wait_for_sync
+   matrix_join_room_and_wait_for_sync
+   matrix_leave_room_and_wait_for_sync
+   matrix_invite_user_to_room_and_wait_for_sync
+   matrix_put_room_state_and_wait_for_sync
+   matrix_advance_room_receipt_and_wait_for_sync
+   sync_timeline_contains
+);
+
+sub matrix_do_and_wait_for_sync
+{
+   my ( $user, %params ) = @_;
+
+   my $check = delete $params{check} or die "Must supply a 'check' param";
+   my $do = delete $params{do} or die "Must supply a 'do' param";
+   $params{timeout} = $params{timeout} // 1000;
+
+   my $next_batch;
+
+   matrix_sync( $user,
```

Comment:
Could do with a small amount of documentation to explain what the `do` and `check` functions do; especially with the two arguments to `check`.